### DISCUSSION TITLE
Data-importer, logs, permission denied

### DISCUSSION THREAD

**Discussion by th3guard1an** [firefly-iii/firefly-iii#7543] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Acessing firefly3 data importer website gives the following error in red.

### Debug information

UnexpectedValueException:
The stream or file "/var/www/html/data-importer/storage/logs/data-import-2023-05-23.log" could not be opened in append mode: Failed to open stream: Permission denied
The exception occurred while attempting to log: The stream or file "/var/www/html/data-importer/storage/logs/data-import-2023-05-23.log" could not be opened in append mode: Failed to open stream: Permission denied
...
Context: {"exception":{}}

  at /var/www/html/data-importer/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php:134
  at Monolog\Handler\StreamHandler->write()

### Additional info

I've a crontab script running in the background to import bank data, it's being run on a user crontab, without sudo commands and is working fine, even while the website is down.

 I've tried to simply alter the file, and on last try to "chmod -R 777 /var/www/html/data-importer" the entire directory, and this resolves the issue for the day, but it restarts on the nexts day log (so when a new .log file is generated by the data-importer). 
I've installed data-importer while on root.
This is the latest version, installed on a proxmox Ubuntu 22.04 Jammy VM environment.

**Comment by th3guard1an** [comment]
I followed this guide:
https://gist.github.com/Engr-AllanG/e87f827092e3a6b876b912cd897428ae

and ran:

sudo chown -R www-data:www-data data-importer
sudo chmod -R 775 data-importer/storage

on install, and it worked fine for a few days,

**Comment by JC5** [comment]
Under which user is the cron job running?

**Reply by th3guard1an** [reply]
I created a user named "olivia", and once "su olivia", I run "crontab -e", and have "@hourly php /var/www/html/data-importer/artisan importer:auto-import /var/www/html/data-importer/auto-import"
Although at first, I was running it on root's crontab, and changed it, as a diagnose step.

**Reply by JC5** [reply, marked as answer]
Try running it under user `www-data`, that's the user in the container.

**Comment by th3guard1an** [comment]
Seems to be working correctly. Thank you.
